src/main/java/service/strategy/strategyimpl/SupplyOperation.java

```diff
@@ -0,0 +1,13 @@
+package service.strategy.strategyimpl;
+
+import dao.Storage;
+import model.FruitRecord;
+
+public class SupplyOperation implements OperationHandler{
+
+    @Override
+    public void apply(FruitRecord transaction) {
+        int supplyQuantity = transaction.getQuantity();
+        Storage.storage.merge(transaction.getFruit(), supplyQuantity, Integer::sum);
```

Comment:
While using `merge` from the `Map` interface is correct, ensure that your storage implementation is thread-safe if you plan to operate in a multi-threaded environment. If thread safety is a concern, consider using `ConcurrentHashMap` or adding appropriate synchronization.